remotes/docker/config/hosts.go

```diff
@@ -0,0 +1,481 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// config package containers utilities for helping configure the Docker resolver
+package config
+
+import (
+	"context"
+	"crypto/tls"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/pkg/errors"
+)
+
+type hostConfig struct {
```

Comment:
Hi, if cri plugin want to use some tls configuration, cri plugin can not reach here.

I think maybe cri plugin also need this feature, and plugin just use WithSkipVerify(args) or WithHostDir(args) to finish

Reply:
The CRI plugin already has its own way of turning its configuration into a registry host list. Instead of trying to expand the CRI configuration to support everything this does, I would rather allow CRI plugin to specify a hosts directory so containerd doesn't require a restart to pick up registry configuration changes. If we decide it is best to share this code in CRI configuration, we can export it then, that might not be the best route though.

Reply:
With users moving from dockershim I think this would be pretty desirable. Has there been any follow-up on this?